File: typescript/src/ui/overlays/new-life.ts
import Colors from '../colors';
import Fonts from '../fonts';
import Overlay from './overlay';

export default class NewLife extends Overlay {
    private pastMillis: number = 0;
    private displayNewLife = true;

    draw() {
        const now = this.p5.millis();

        if (now - this.pastMillis > 500) {
            this.displayNewLife = !this.displayNewLife;
            this.pastMillis = now;
        }

        if (this.displayNewLife) {
            this.p5.push();

            this.p5.textFont(Fonts.fontThin);
            this.p5.fill(Colors.EDGE);
            this.p5.textFont(Fonts.fontLight);
            this.p5.textAlign(this.p5.CENTER);
            this.p5.textSize(20);

            this.p5.text('NEW LIFE!', this.centerX, 85);

            this.p5.pop();
        }
    }
}

File: rust/src/overlays/new_life.rs
use web_sys::CanvasRenderingContext2d;

use crate::{sprites::sprite::CanvasDimension, utils::millis};

pub struct NewLife {
    past_millis: f64,
    display_new_life: bool,
    center_x: f64,
}

impl NewLife {
    pub fn new(canvas: CanvasDimension) -> NewLife {
        NewLife {
            past_millis: 0.0,
            display_new_life: false,
            center_x: canvas.width / 2.0,
        }
    }
}

impl NewLife {
    pub fn update(&mut self) {
        let now = millis();

        if now - self.past_millis > 1000.0 {
            self.display_new_life = !self.display_new_life;
            self.past_millis = now;
        }
    }

    pub fn draw(&self, canvas: CanvasRenderingContext2d) {
        if self.display_new_life {
            canvas.save();

            canvas.set_text_align("center");
            canvas.set_font("300 20px 'Exo 2'");
            let _result = canvas.fill_text("NEW LIFE!", self.center_x, 85.0);

            canvas.restore();
        }
    }
}
